lib/rules/preserve-caught-error.js

```diff
@@ -0,0 +1,263 @@
+
+		//----------------------------------------------------------------------
+		// Helpers
+		//----------------------------------------------------------------------
+
+		/**
+		 * Recursively searches for all the `ThrowStatement` nodes within a block.
+		 * @param {ASTNode} blockStatement `BlockStatement` to be searched.
+		 * @returns {ASTNode[]} An array of `ThrowStatement` nodes found.
+		 */
+		function findThrowStatements(blockStatement) {
+			const throws = [];
+
+			for (const node of blockStatement.body) {
+				if (node.type === "ThrowStatement") {
+					throws.push(node);
+				}
+
+				// Handle if blocks
+				if (node.type === "IfStatement") {
+					// If block with braces
+					if (node.consequent.type === "BlockStatement") {
+						throws.push(...findThrowStatements(node.consequent));
+					}
+					// Single statement without braces
+					else if (node.consequent.type === "ThrowStatement") {
+						// e.g., if (cond) throw new Error("...");
+						throws.push(node.consequent);
+					}
+
+					// Handle else blocks (if present)
+					if (node.alternate) {
+						if (node.alternate.type === "BlockStatement") {
+							throws.push(...findThrowStatements(node.alternate));
+						} else if (node.alternate.type === "ThrowStatement") {
+							throws.push(node.alternate);
+						}
+					}
+				}
+
+				// Handle switch cases
+				else if (node.type === "SwitchStatement") {
+					for (const switchCase of node.cases) {
+						for (const statement of switchCase.consequent) {
+							if (statement.type === "ThrowStatement") {
+								throws.push(statement);
+							} else if (statement.type === "BlockStatement") {
+								throws.push(...findThrowStatements(statement));
+							}
+						}
+					}
+				}
+
+				// General fallback for loops etc.
+				else if (
+					"body" in node &&
+					node.body &&
+					"type" in node.body &&
+					node.body.type === "BlockStatement"
+				) {
+					throws.push(...findThrowStatements(node.body));
+				}
+			}
+
+			return throws;
+		}
+
+		/**
+		 * Checks if a `ThrowStatement` is constructing and throwing a new `Error` object.
+		 *
+		 * Covers all the error types on `globalThis` that support `cause` property:
+		 * https://github.com/microsoft/TypeScript/blob/main/src/lib/es2022.error.d.ts
+		 * @param {ASTNode} throwStatement The `ThrowStatement` that needs to be checked.
+		 * @param {string[]} customErrorTypes A list of custom error types that need to be considered as well.
+		 * @returns {boolean} `true` if a new "Error" is being thrown, else `false`.
+		 */
+		function isThrowingNewError(throwStatement, customErrorTypes = []) {
+			const builtInErrorTypes = new Set([
+				"Error",
+				"EvalError",
+				"RangeError",
+				"ReferenceError",
+				"SyntaxError",
+				"TypeError",
+				"URIError",
+				"AggregateError",
+			]);
+
+			const allErrorTypes = new Set([
+				...builtInErrorTypes,
+				...customErrorTypes,
+			]);
+			return (
+				throwStatement.argument.type === "NewExpression" &&
```

Comment:
Besides `NewExpression` it would be good to consider also `CallExpression` nodes in this check, in order to report on this case:

```js
try { doSomething(); } catch (err) { throw Error("Something failed"); }
```

All native JavaScript errors can be constructed without the `new` syntax.